src/SongPro.js

```diff
@@ -74,33 +84,62 @@ export class SongPro {
 
     const line = new Line();
 
-    let captures = this.scan(text, /(\[[\w#b/]+])?([\w\s',.!()_\-"]*)/gi);
-    captures = flatten(captures);
-    const groups = chunk(captures, 2);
+    if (text.startsWith("|-")) {
+      line.tablature = text;
+    } else if (text.startsWith("| ")) {
+      let captures = this.scan(text, MEASURES_REGEX);
+      captures = flatten(captures);
 
-    for (let i = 0; i < groups.length; i++) {
-      const group = groups[i];
-      let chord = group[0];
-      let lyric = group[1];
+      const measures = [];
 
-      const part = new Part();
+      for (let i = 0; i < captures.length; i++) {
```

Comment:
see below for for..of loop style iteration

Reply:
Fixed!